### PR TITLE
DAOS-623 codeowners: add client owner

### PR DESCRIPTION
**Pull request by mchaarawi**
Skip-build: true
Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -15,3 +15,7 @@
 
 # Jenkinsfile changes should be reviewed by Release Engineering
 #Jenkinsfile @daos-stack/release-engineering
+
+# any PR that touches client API or high level client code
+#src/client @daos-stack/owner-client-ap
+#include/daos_* @daos-stack/owner-client-api
```
Comment (jolivier23) [inline .github/CODEOWNERS:21]: Per the discussion, I think it would be client-api-watcher

**Review by jolivier23 — CHANGES_REQUESTED** [review]

**Review by jolivier23 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -15,3 +15,7 @@
 
 # Jenkinsfile changes should be reviewed by Release Engineering
 #Jenkinsfile @daos-stack/release-engineering
+
+# any PR that touches client API or high level client code
+#src/client @daos-stack/client-api-watcher
```
Comment (jolivier23) [inline .github/CODEOWNERS:20]: Is it your intent not to enable this yet?

Comment (mchaarawi) [reply .github/CODEOWNERS:20]: i saw everything is disabled, so didn't know if i should start enabling this now.

**Review by jolivier23 — APPROVED** [review]
Dd

**Pull request closed without merging**